lib/axon.ex

```diff
@@ -1644,10 +1644,34 @@ defmodule Axon do
   end
 
   @doc """
-  Splits input graph into a container of `n` input graphs
-  along the given axis.
+  Splits input graph into a container of `m` other input graphs 
+  In case `n` is a number the input graph is split in n slices with the same size
+  In case `n` is a list the input graph is split in Enum.count slices with the 
+  size of the i-th element of the list.
   """
-  def split(%Axon{output_shape: shape} = parent, n, opts \\ []) do
+
+  def split(parent, n, opts \\ []) 
+
+  def split(%Axon{output_shape: shape} = parent, n, opts) when is_list(n) do
+    axis = opts[:axis] || -1
+    names = opts[:name] || []
```

Comment:
```suggestion
  def split(%Axon{name: parent_name_fn, output_shape: shape} = parent, n, opts) when is_list(n) do
    axis = opts[:axis] || -1
    names = opts[:name]
    names =
      case names do
        nil ->
          for i <- 0..n - 1 do
            fn op, op_counts -> parent_name_fn.(op, op_counts) <> "_i" end
          end
        
        names ->
          names
      end
```

This will give you names, `layer` should not be expecting a string in the most recent Axon :)

Reply:
Thank you Sean for this suggestion. The problem is that at rows 120-125 of axon.ex into the current main branch we have:

```
  @doc type: :special
  def layer(parent, op, output_shape, parameters, name \\ nil, opts \\ [])
      when is_atom(op) or (is_function(op) and is_map(parameters)) do
    op_name = if is_atom(op), do: op, else: :layer

    {id, name} = unique_identifiers(op_name, name)
```
So it seems layer function is still expecting name as a string and not as a function. Should I add a check on name type into the layer function?